src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -410,29 +380,11 @@ private string GenerateForCollection(TypeGenerationSpec typeGenerationSpec)
                 TypeGenerationSpec? collectionKeyTypeMetadata = typeGenerationSpec.CollectionKeyTypeMetadata;
                 Debug.Assert(!(typeGenerationSpec.ClassType == ClassType.Dictionary && collectionKeyTypeMetadata == null));
                 string? keyTypeCompilableName = collectionKeyTypeMetadata?.TypeRef;
-                string? keyTypeReadableName = collectionKeyTypeMetadata?.TypeInfoPropertyName;
```

Comment:
What is this being replaced by in generated code?

Reply:
we use converter to get key type and resolver to get type info rather than directly assigning type info.